src/lib/network_pool/test_utils.ml

```diff
@@ -0,0 +1,831 @@
+      module Breadcrumb = struct
+        type t = Mock_staged_ledger.t
+
+        let staged_ledger = Fn.id
+      end
+
+      type best_tip_diff =
+        { new_commands : User_command.Valid.t With_status.t list
+        ; removed_commands : User_command.Valid.t With_status.t list
+        ; reorg_best_tip : bool
+        }
+
+      type t = best_tip_diff Broadcast_pipe.Reader.t * Breadcrumb.t ref
+
+      let create ?permissions :
+          unit -> t * best_tip_diff Broadcast_pipe.Writer.t =
+       fun () ->
+        let zkappify_account (account : Account.t) : Account.t =
+          let zkapp =
+            Some { Zkapp_account.default with verification_key = Some vk }
+          in
+          { account with
+            zkapp
+          ; permissions =
+              ( match permissions with
+              | Some p ->
+                  p
+              | None ->
+                  Permissions.user_default )
+          }
+        in
+        let pipe_r, pipe_w =
+          Broadcast_pipe.create
+            { new_commands = []; removed_commands = []; reorg_best_tip = false }
+        in
+        let initial_balance =
+          Currency.Balance.of_mina_string_exn "900000000.0"
+        in
+        let ledger = Mina_ledger.Ledger.create_ephemeral ~depth:10 () in
+        Array.iteri test_keys ~f:(fun i kp ->
+            let account_id =
+              Account_id.create
+                (Public_key.compress kp.public_key)
+                Token_id.default
+            in
+            let _tag, account, loc =
+              Or_error.ok_exn
+              @@ Mina_ledger.Ledger.Ledger_inner.get_or_create ledger account_id
+            in
+            (* set the account balance *)
+            let account = { account with balance = initial_balance } in
+            (* zkappify every other account *)
+            let account =
+              if i mod 2 = 0 then account else zkappify_account account
+            in
+            Mina_ledger.Ledger.Ledger_inner.set ledger loc account ) ;
+        ((pipe_r, ref ledger), pipe_w)
+
+      let best_tip (_, best_tip) = !best_tip
+
+      let best_tip_diff_pipe (pipe, _) = pipe
+    end
+
+    module Test =
+      Make0 (Mock_base_ledger) (Mock_staged_ledger) (Mock_transition_frontier)
+
+    type test =
+      { txn_pool : Test.Resource_pool.t
+      ; best_tip_diff_w :
+          Mock_transition_frontier.best_tip_diff Broadcast_pipe.Writer.t
+      ; best_tip_ref : Mina_ledger.Ledger.t ref
+      ; frontier_pipe_w :
+          Mock_transition_frontier.t option Broadcast_pipe.Writer.t
+      }
+
+    let pool_max_size = 25
+
+    let apply_initial_ledger_state t init_ledger_state =
+      let new_ledger =
+        Mina_ledger.Ledger.create_ephemeral
+          ~depth:(Mina_ledger.Ledger.depth !(t.best_tip_ref))
+          ()
+      in
+      Mina_ledger.Ledger.apply_initial_ledger_state new_ledger init_ledger_state ;
+      t.best_tip_ref := new_ledger
+
+  let ledger_snapshot t =
```

Comment:
Suggest to inline it into `src/lib/network_pool/transacation_pool_reorg_tests.ml` and return `Account_spec.t` instead of a tuple